docs(mesh): add comments to example Vault mTLS config (#3541)
Kong/docs.konghq.com: app/mesh/1.5.x/features/vault.md

            pki: kmesh-pki-default # name of the configured PKI
            role: dataplane-proxies # name of the role that will be used to generate data plane proxy certificates
            commonName: {% raw %}'{{ tag "kuma.io/service" }}.mesh'{% endraw %} # optional. If set, then commonName is added to the certificate. You can use "tag" directive to pick a tag which will be base for commonName.

            tls: # options for connecting to Vault via TLS
              skipVerify: false   # if set to true, caCert is optional, should only be used in development
              caCert:             # caCert is used to verify the TLS certificate presented by Vault

>                 secret: sec-1     # can be file, secret or inline

Reviewer: `can be secret, inline or inlineString`. While file is technically valid, our reconciliation process does not react on changes in files. If we don't have validation for that, we should introduce it

Reviewer 2: ```suggestion
                secret: sec-1     # one of secret, inline, or inlineString
```

>               serverName: ""      # the SNI to use when connecting to Vault

Reviewer: `optional: the SNI to use when connecting to Vault`

Reviewer 2: ```suggestion
              serverName: ""      # optional. The SNI to use when connecting to Vault
```

            auth: # how to authenticate Kong Mesh when connecting to Vault
              token:

>                 secret: token-1  # can be file, secret or inlineString

Reviewer 2: ```suggestion
                secret: token-1  # one of secret, inline, or inlineString
```
